exporters/metric/cortex/cortex.go

```diff
@@ -86,11 +109,228 @@ func InstallNewPipeline(config Config, options ...push.Option) (*push.Controller
 	return pusher, nil
 }
 
-// addHeaders adds required headers as well as all headers in Header map to a http
-// request.
+// ConvertToTimeSeries converts a CheckpointSet to a slice of TimeSeries pointers
+func (e *Exporter) ConvertToTimeSeries(checkpointSet export.CheckpointSet) ([]*prompb.TimeSeries, error) {
+	var aggError error
+	var timeSeries []*prompb.TimeSeries
+
+	// Iterate over each record in the checkpoint set and convert to TimeSeries
+	aggError = checkpointSet.ForEach(e, func(record metric.Record) error {
+		// Convert based on aggregation type
+		agg := record.Aggregation()
+
+		// Check if aggregation has Sum value
+		if sum, ok := agg.(aggregation.Sum); ok {
+			tSeries, err := convertFromSum(record, sum)
+			if err != nil {
+				return err
+			}
+
+			timeSeries = append(timeSeries, tSeries)
+		}
+
+		// Check if aggregation has MinMaxSumCount value
+		if minMaxSumCount, ok := agg.(aggregation.MinMaxSumCount); ok {
+			tSeries, err := convertFromMinMaxSumCount(record, minMaxSumCount)
+			if err != nil {
+				return err
+			}
+
+			timeSeries = append(timeSeries, tSeries...)
+		} else if lastValue, ok := agg.(aggregation.LastValue); ok {
+			tSeries, err := convertFromLastValue(record, lastValue)
+			if err != nil {
+				return err
+			}
+
+			timeSeries = append(timeSeries, tSeries)
+		}
```

Comment:
Should we catch this case where no match was found above?

Reply:
What would the expected behavior be if no aggregations are matched? I assume that means the user provided their own and we should log that the record wasn't converted?

Also to note, I'll be adding in distribution and histogram in another PR soon.